### ISSUE TITLE
Dict[str, int] not assignable to Mapping[Union[str, bytes], int]

### ISSUE THREAD

**Issue by jmorag** [microsoft/pyright#936]
**Describe the bug**
Attempting to pass a variable of type `Dict[str, int]` to a function expecting an argument of type `Mapping[Union[str, bytes], int]` produces the error
```
5:3 - error: Argument of type "dict[str, int]" cannot be assigned to parameter "mapping" of type "Mapping[str | bytes, int]" in function "f"
  TypeVar "_KT" is invariant
    Type "str" cannot be assigned to type "str | bytes" (reportGeneralTypeIssues)
```

**To Reproduce**
```python
from typing import Mapping, Dict, Union
def f(mapping: Mapping[Union[str, bytes], int]):
    return mapping
f({"x": 1})
f({b"x": 1})
```
```
  4:3 - error: Argument of type "dict[str, int]" cannot be assigned to parameter "mapping" of type "Mapping[str | bytes, int]" in function "f"
  TypeVar "_KT" is invariant
    Type "str" cannot be assigned to type "str | bytes" (reportGeneralTypeIssues)
  5:3 - error: Argument of type "dict[bytes, int]" cannot be assigned to parameter "mapping" of type "Mapping[str | bytes, int]" in function "f"
  TypeVar "_KT" is invariant
    Type "bytes" cannot be assigned to type "str | bytes" (reportGeneralTypeIssues)
2 errors, 0 warnings
```
**Expected behavior**
`Dict[str, int]` should unify with `Mapping[Union[str, bytes], int]`. 

**VS Code extension or command-line**
Command line
```
> pyright --version
pyright 1.1.62
```
**Additional context**
Both `g` and `h` shown below typecheck correctly, so I imagine that the error has to do with `Dict`/`Mapping`. If so, the error message is misleading.
```python
def g(mapping: Dict[Union[str, bytes], int]):
    return mapping
g({"x": 1})
g({b"x": 1})

def h(x: Union[str, bytes]):
    return x
h(b"x")
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next version of Pyright.

**Comment by erictraut** [comment]
This is now addressed in version 1.1.63 of Pyright, which I just published. It will also be in the next release of Pylance.

**Comment by jmorag** [comment]
Just upgraded and the error is fixed. Thank you very much!
